### PR TITLE
Enable PSA_CRYPTO_CONFIG in test_psa_crypto_drivers

### LINKED ISSUE #9658: Enable MBEDTLS_PSA_CRYPTO_CONFIG in test_psa_crypto_drivers
### Suggested enhancement
With the components relying on depends.py, test_psa_crypto_drivers is the last component not adapted to MBEDTLS_PSA_CRYPTO_CONFIG being enabled. This task is to enable MBEDTLS_PSA_CRYPTO_CONFIG in
test_psa_crypto_drivers and eventually have the same testing as with MBEDTLS_PSA_CRYPTO_CONFIG
disabled.

### Justification
Mbed TLS needs this for the repo split.

### PR DESCRIPTION
**Pull request by ronald-cron-arm**
## Description
Fixes #9658

## PR checklist
- [x] **changelog** not required because: the PR just change how a test component is implemented 
- [x] **development PR** here
- [x] **framework PR** not required
- [x] **3.6 PR** not required because: 4.x only
- [x] **2.28 PR** not required because: 4.x only 
- **tests**  not required because: just adapting a test component

### CODE REVIEW

**Comment by ronald-cron-arm** [comment]
I have run the test_psa_crypto_drivers component on development (as of cb81d8086f) and on the head of the PR. The result of ```comm -23 <(cat outcome-dev-cb81d8086f.csv | sort) <(cat outcome-360928af9d.csv | sort) | grep ";PASS;"``` (test cases run on development and not on the tip of this PR) on the outcome.csv files is:

Linux-x86_64;component_test_psa_crypto_drivers;test_suite_config.mbedtls_boolean;Config: !MBEDTLS_PSA_CRYPTO_CONFIG;PASS;
Linux-x86_64;component_test_psa_crypto_drivers;test_suite_config.psa_boolean;Config: !PSA_WANT_ALG_PBKDF2_AES_CMAC_PRF_128;PASS;
Linux-x86_64;component_test_psa_crypto_drivers;test_suite_config.psa_boolean;Config: !PSA_WANT_ALG_PBKDF2_HMAC;PASS;
Linux-x86_64;component_test_psa_crypto_drivers;test_suite_psa_crypto_op_fail.generated;PSA key_derivation PBKDF2_AES_CMAC_PRF_128: !PBKDF2_AES_CMAC_PRF_128;PASS;
Linux-x86_64;component_test_psa_crypto_drivers;test_suite_psa_crypto_op_fail.generated;PSA key_derivation PBKDF2_HMAC(MD5): !PBKDF2_HMAC;PASS;
Linux-x86_64;component_test_psa_crypto_drivers;test_suite_psa_crypto_op_fail.generated;PSA key_derivation PBKDF2_HMAC(RIPEMD160): !PBKDF2_HMAC;PASS;
Linux-x86_64;component_test_psa_crypto_drivers;test_suite_psa_crypto_op_fail.generated;PSA key_derivation PBKDF2_HMAC(SHA_1): !PBKDF2_HMAC;PASS;
Linux-x86_64;component_test_psa_crypto_drivers;test_suite_psa_crypto_op_fail.generated;PSA key_derivation PBKDF2_HMAC(SHA_224): !PBKDF2_HMAC;PASS;
Linux-x86_64;component_test_psa_crypto_drivers;test_suite_psa_crypto_op_fail.generated;PSA key_derivation PBKDF2_HMAC(SHA_256): !PBKDF2_HMAC;PASS;
Linux-x86_64;component_test_psa_crypto_drivers;test_suite_psa_crypto_op_fail.generated;PSA key_derivation PBKDF2_HMAC(SHA3_224): !PBKDF2_HMAC;PASS;
Linux-x86_64;component_test_psa_crypto_drivers;test_suite_psa_crypto_op_fail.generated;PSA key_derivation PBKDF2_HMAC(SHA3_256): !PBKDF2_HMAC;PASS;
Linux-x86_64;component_test_psa_crypto_drivers;test_suite_psa_crypto_op_fail.generated;PSA key_derivation PBKDF2_HMAC(SHA3_384): !PBKDF2_HMAC;PASS;
Linux-x86_64;component_test_psa_crypto_drivers;test_suite_psa_crypto_op_fail.generated;PSA key_derivation PBKDF2_HMAC(SHA3_512): !PBKDF2_HMAC;PASS;
Linux-x86_64;component_test_psa_crypto_drivers;test_suite_psa_crypto_op_fail.generated;PSA key_derivation PBKDF2_HMAC(SHA_384): !PBKDF2_HMAC;PASS;
Linux-x86_64;component_test_psa_crypto_drivers;test_suite_psa_crypto_op_fail.generated;PSA key_derivation PBKDF2_HMAC(SHA_512): !PBKDF2_HMAC;PASS;
Linux-x86_64;component_test_psa_crypto_drivers;test_suite_psa_crypto_op_fail.generated;PSA sign RSA_PSS_ANY_SALT(MD5): !RSA_PSS_ANY_SALT with RSA_KEY_PAIR;PASS;
Linux-x86_64;component_test_psa_crypto_drivers;test_suite_psa_crypto_op_fail.generated;PSA sign RSA_PSS_ANY_SALT(MD5): !RSA_PSS_ANY_SALT with RSA_PUBLIC_KEY;PASS;
Linux-x86_64;component_test_psa_crypto_drivers;test_suite_psa_crypto_op_fail.generated;PSA sign RSA_PSS_ANY_SALT(RIPEMD160): !RSA_PSS_ANY_SALT with RSA_KEY_PAIR;PASS;
Linux-x86_64;component_test_psa_crypto_drivers;test_suite_psa_crypto_op_fail.generated;PSA sign RSA_PSS_ANY_SALT(RIPEMD160): !RSA_PSS_ANY_SALT with RSA_PUBLIC_KEY;PASS;
Linux-x86_64;component_test_psa_crypto_drivers;test_suite_psa_crypto_op_fail.generated;PSA sign RSA_PSS_ANY_SALT(SHA_1): !RSA_PSS_ANY_SALT with RSA_KEY_PAIR;PASS;
Linux-x86_64;component_test_psa_crypto_drivers;test_suite_psa_crypto_op_fail.generated;PSA sign RSA_PSS_ANY_SALT(SHA_1): !RSA_PSS_ANY_SALT with RSA_PUBLIC_KEY;PASS;
Linux-x86_64;component_test_psa_crypto_drivers;test_suite_psa_crypto_op_fail.generated;PSA sign RSA_PSS_ANY_SALT(SHA_224): !RSA_PSS_ANY_SALT with RSA_KEY_PAIR;PASS;
Linux-x86_64;component_test_psa_crypto_drivers;test_suite_psa_crypto_op_fail.generated;PSA sign RSA_PSS_ANY_SALT(SHA_224): !RSA_PSS_ANY_SALT with RSA_PUBLIC_KEY;PASS;
Linux-x86_64;component_test_psa_crypto_drivers;test_suite_psa_crypto_op_fail.generated;PSA sign RSA_PSS_ANY_SALT(SHA_256): !RSA_PSS_ANY_SALT with RSA_KEY_PAIR;PASS;
Linux-x86_64;component_test_psa_crypto_drivers;test_suite_psa_crypto_op_fail.generated;PSA sign RSA_PSS_ANY_SALT(SHA_256): !RSA_PSS_ANY_SALT with RSA_PUBLIC_KEY;PASS;
Linux-x86_64;component_test_psa_crypto_drivers;test_suite_psa_crypto_op_fail.generated;PSA sign RSA_PSS_ANY_SALT(SHA3_224): !RSA_PSS_ANY_SALT with RSA_KEY_PAIR;PASS;
Linux-x86_64;component_test_psa_crypto_drivers;test_suite_psa_crypto_op_fail.generated;PSA sign RSA_PSS_ANY_SALT(SHA3_224): !RSA_PSS_ANY_SALT with RSA_PUBLIC_KEY;PASS;
Linux-x86_64;component_test_psa_crypto_drivers;test_suite_psa_crypto_op_fail.generated;PSA sign RSA_PSS_ANY_SALT(SHA3_256): !RSA_PSS_ANY_SALT with RSA_KEY_PAIR;PASS;
Linux-x86_64;component_test_psa_crypto_drivers;test_suite_psa_crypto_op_fail.generated;PSA sign RSA_PSS_ANY_SALT(SHA3_256): !RSA_PSS_ANY_SALT with RSA_PUBLIC_KEY;PASS;
Linux-x86_64;component_test_psa_crypto_drivers;test_suite_psa_crypto_op_fail.generated;PSA sign RSA_PSS_ANY_SALT(SHA3_384): !RSA_PSS_ANY_SALT with RSA_KEY_PAIR;PASS;
Linux-x86_64;component_test_psa_crypto_drivers;test_suite_psa_crypto_op_fail.generated;PSA sign RSA_PSS_ANY_SALT(SHA3_384): !RSA_PSS_ANY_SALT with RSA_PUBLIC_KEY;PASS;
Linux-x86_64;component_test_psa_crypto_drivers;test_suite_psa_crypto_op_fail.generated;PSA sign RSA_PSS_ANY_SALT(SHA3_512): !RSA_PSS_ANY_SALT with RSA_KEY_PAIR;PASS;
Linux-x86_64;component_test_psa_crypto_drivers;test_suite_psa_crypto_op_fail.generated;PSA sign RSA_PSS_ANY_SALT(SHA3_512): !RSA_PSS_ANY_SALT with RSA_PUBLIC_KEY;PASS;
Linux-x86_64;component_test_psa_crypto_drivers;test_suite_psa_crypto_op_fail.generated;PSA sign RSA_PSS_ANY_SALT(SHA_384): !RSA_PSS_ANY_SALT with RSA_KEY_PAIR;PASS;
Linux-x86_64;component_test_psa_crypto_drivers;test_suite_psa_crypto_op_fail.generated;PSA sign RSA_PSS_ANY_SALT(SHA_384): !RSA_PSS_ANY_SALT with RSA_PUBLIC_KEY;PASS;
Linux-x86_64;component_test_psa_crypto_drivers;test_suite_psa_crypto_op_fail.generated;PSA sign RSA_PSS_ANY_SALT(SHA_512): !RSA_PSS_ANY_SALT with RSA_KEY_PAIR;PASS;
Linux-x86_64;component_test_psa_crypto_drivers;test_suite_psa_crypto_op_fail.generated;PSA sign RSA_PSS_ANY_SALT(SHA_512): !RSA_PSS_ANY_SALT with RSA_PUBLIC_KEY;PASS;

Thus only test cases where a feature wasn't enabled before and is enabled now. When we move from the legacy to PSA configuration we usually get in addition the  PBKDF2_HMAC and PBKDF2_AES_CMAC_PRF_128  algorithms (can be enabled only through PSA config). Otherwise, PSA_WANT_ALG_RSA_PSS_ANY_SALT is now defined in crypto_adjust_config_synonyms.h because now PSA_WANT_ALG_RSA_PSS is defined when crypto_adjust_config_synonyms.h is included but was not before (in dev it is defined by config_adjust_psa_from_legacy.h further down in the sequence of header inclusions in build_info.h).

Thus , it seems reasonable for the above test cases to not be run anymore as part of test_psa_crypto_drivers. 
It is reasonable to expect that the test cases above are run by other all.sh components.

File: tf-psa-crypto/drivers/builtin/include/mbedtls/config_adjust_test_accelerators.h [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+ *   test_suite_psa_crypto_driver_wrappers test suite: test of the dispatch to
+ *   drivers and fallbacks.
+ * . this file can be removed.
+ *
+ * This header is used as part of a build containing all the built-in drivers
+ * and all the transparent test drivers as wrappers around the built-in
+ * drivers. All the built-in drivers and the transparent test drivers are
+ * included in the build by starting from a full configuration (config.py full)
+ * and defining PSA_CRYPTO_DRIVER_TEST when building
+ * (make -DPSA_CRYPTO_DRIVER_TEST ...).
```
Comment (mpg) [inline tf-psa-crypto/drivers/builtin/include/mbedtls/config_adjust_test_accelerators.h:23]: This should be `make CFLAGS=...`.

File: tests/scripts/components-configuration-crypto.sh [hunk 2]
```diff
@@ -2673,9 +2673,7 @@ component_test_full_static_keystore () {
 component_test_psa_crypto_drivers () {
     msg "build: full + test drivers dispatching to builtins"
     scripts/config.py full
-    scripts/config.py unset MBEDTLS_PSA_CRYPTO_CONFIG
-    loc_cflags="$ASAN_CFLAGS -DPSA_CRYPTO_DRIVER_TEST_ALL"
-    loc_cflags="${loc_cflags} '-DMBEDTLS_USER_CONFIG_FILE=\"../tests/configs/user-config-for-test.h\"'"
+    loc_cflags="$ASAN_CFLAGS -DPSA_CRYPTO_DRIVER_TEST -DMBEDTLS_CONFIG_ADJUST_TEST_ACCELERATORS"
     loc_cflags="${loc_cflags} -I../tests/include -O2"
```
Comment (mpg) [inline tests/scripts/components-configuration-crypto.sh:2686]: Minor, pre-existing: `-O2` is redundant, it's already in `ASAN_CFLAGS`.

**Review by mpg — DISMISSED** [review]
LGTM, thanks!

**Review by mpg — APPROVED** [review]
LGTM!

**Review by minosgalanakis — APPROVED** [review]
LGTM

**Pull request merged**